app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -320,6 +329,9 @@ private void saveRecording() {
         if (listener != null) {
             listener.onRecordingCompletion(fileName);
         }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            unregisterAudioRecordingConfigurationChangeCallback();

```

Comment:
Would this callback wil never get unregistered if user don't call save() ?

Reply:
I think once the recording starts, we [can't dismiss the dialog](https://github.com/dimagi/commcare-android/blob/2df74444f4fce60f3c0545022fab8f837c73b90b/app/src/org/commcare/views/widgets/RecordingFragment.java#L189C9-L189C25) nor [go back](https://github.com/dimagi/commcare-android/blob/2df74444f4fce60f3c0545022fab8f837c73b90b/app/src/org/commcare/views/widgets/RecordingFragment.java#L169) without saving.

Reply:
got it, can we also check if the callback works after screen rotation i.e rotate the screen after you start recording and then change audio config and see if the callback still gets triggered?

Reply:
screen rotation is disabled in [onCreateView()](https://github.com/dimagi/commcare-android/blob/2df74444f4fce60f3c0545022fab8f837c73b90b/app/src/org/commcare/views/widgets/RecordingFragment.java#L81) and [startRecording()](https://github.com/dimagi/commcare-android/blob/2df74444f4fce60f3c0545022fab8f837c73b90b/app/src/org/commcare/views/widgets/RecordingFragment.java#L168)